### DISCUSSION TITLE
Garmin icons in favorites

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#529] [Q&A] [answered]
Waypoints:  Free individual locations as are clearly definied in the gpx standard as being gpx (wpt) waypoints !

The use of Garmin (png) icons triggered by the gpx wpt sym are widespread. By the wpt gpx sym many apps already do show the associated Garmin or compatible icons. Optimally the Cruiser app should offer this in the same way. Even with png icons, the presentation is very neat.

A straightforward simple way via sym usage is desirable. Provided less resistance to png icons something like this is definitely possible.

![image](https://github.com/devemux86/cruiser/assets/54167201/c7eec1bf-e07d-4852-9015-c09b57d40f8f)

The popular range of Garmin icons (png) is supported by default in apps like Locus map.

Proposal Cruiser.

![image](https://github.com/user-attachments/assets/b7fec35d-f653-4380-8dce-09aa9078bd5c)

Example file gpx.
[Cape_Cod_wpt_leg1_leg2_loop_option_navtrk.gpx.txt](https://github.com/user-attachments/files/23630496/Cape_Cod_wpt_leg1_leg2_loop_option_navtrk.gpx.txt)
Result.
[Cape_Cod_wpt_leg1_leg2_loop_option_navtrk-map.html](https://github.com/user-attachments/files/23630515/Cape_Cod_wpt_leg1_leg2_loop_option_navtrk-map.html)

- **Garmin Icons list**.
[Garmin Icons-48x.zip](https://github.com/user-attachments/files/25098073/Garmin.Icons-48x.zip)

_This file contains the 266 icons from Garmin Basecamp.
The official Garmin list contains 6 gpx sym with a / slash.
Garmin programs only support the gpx sym with a / slash.
In the icon name png the slash is replaced by a - hyphen.
The gpz transfers can only be done by using the - hyphen.
Locus recognizes the gpx sym with a slash or a hyphen._

_Support for an unlimited number of icons is not possible now. However, the sender can easily transfer the matching icon images associated with the gpx wpt sym into a compressed gpz file._  See: https://github.com/devemux86/cruiser/discussions/127

**Comment by MotoUKRider** [comment]
If you can't join them, beat them!
As I was quite regimented in my Waypoint naming early on, I took advantage of that to produce my own solution:

![cruiser-gpx-maker](https://github.com/devemux86/cruiser/assets/125696109/667413ee-9f8f-4a8b-b6b9-562d56271263)

It's hard-coded to specific Cruiser icons+colours, but works just fine.  Here's an early example [from a recent thread](https://github.com/devemux86/cruiser/discussions/495#discussioncomment-9192571):

![favorites-import-test](https://github.com/devemux86/cruiser/assets/125696109/68ceb284-971d-47fc-a550-eb53a076683a)

I have plans to make it more user-friendly so that it's of use to others (if you don't mind a Windoze app), but this is just how I got around a map of orange stars in the short-term :laughing:

**Comment by 0709wiwiwi** [comment]
Good to know. Adde a remark of the 6 Garmin Navaid sym with a / slash in the first post.

**Reply by MotoUKRider** [reply]
@0709wiwiwi As I have found, it really pays to look at the GPX names because the BaseCamp "tooltip" in the Waypoint properties window isn't always accurate!

eg: `Dog Treed` (in GPX output and your files) but `Dog, treed` in BaseCamp tooltip:
![basecamp-icon-names](https://github.com/user-attachments/assets/4f4b912e-aab6-4dac-9b16-db3c75e9af3c)

The / character is understandable, but there seems to be a few instances where a comma is shown, but not exported - but the "Contact" icons have the comma in the tooltip AND the GPX.  Funny ol' day!

**Reply by devemux86** [reply]
@MotoUKRider What is written in the `sym` field is important, because it's supposed to be the icon file name.

**Reply by MotoUKRider** [reply]
@devemux86 Yes, that's the jist of my post - After your e-mail about this I started-off using the tooltip before wondering why I was not getting some groups populated in my little utility.  It turns out that opening the GPX file and having a peek at the `sym` field _really_ helps get the name-triggers working!

This 2nd GPX output works OK but I certainly don't need 200+ items to either match/group/define so I'm sticking with my original method of using the item's name, "Photo, Nice Place" or "Food, Pizza" which saves to a single "Favorite" GPX with half-dozen Cruiser icons regardless of the actual BaseCamp icon.  It makes the name longer (more of an issue in portrait with a large font) and I suppose I could trim the names as the GPX is created but then they'd be different between the 2 systems. Meh.

I quite like that I can build a GPX with a default icon/color at the top, and then populate it with custom entries if they match, but just define the "name" for anything that isn't caught by my filter - resulting in an orange star.  I presently have 1275 items in my Cruiser Favorites, exported from BaseCamp as a single GPX of varying colors/icons.

All this is a bit custom and probably of no interest to the general user(!) but for those who trip-plan on _Solution A_ and navigate with _Solution B_, some extra work is required at this time if you want to "_Plan in comfort, ride in style_" :tm:

**Comment by devemux86** [comment]
Garmin icons as selection in favorites and import from GPX are available for testing in **Cruiser 5.8.0 Beta**:
- https://github.com/devemux86/cruiser/discussions/887

**Comment by MotoUKRider** [comment]
For those interested, my own housekeeping just got a lot more efficient - Unless I explicitly want to use internal Cruiser icons/colours with my BaseCamp Waypoints GPX (now running to ~1400 items) I can just push the exported Garmin GPX into Cruiser without using my little utility.

_**Before:**_
![cruiser-icons-default](https://github.com/user-attachments/assets/c9d558ec-f984-40a5-8243-7f4a59facc1e)

**_Current_**:
![cruiser-icons-garmin](https://github.com/user-attachments/assets/f283d7c2-d8aa-46a5-8908-1fbdf28d94b1)

As this is also in Cruiser Desktop, it makes the ability to plan a route using fast (and nice-looking!) map/s with meaningful icons much easier in some situations compared to BaseCamp.

**Comment by devemux86** [comment, marked as answer]
- Implemented in https://github.com/devemux86/cruiser/discussions/891
